pp.c

```diff
@@ -7700,6 +7700,23 @@ PP(pp_anonconst)
 }
 
 
+/* Helper function for use during signature argument handling from @_.
+ * Replaces elements in an AV with a new SV cloned from the original value
+ * at each position from startix onwards until endix.
+ */
+#define av_refresh_elements_range(av, startix, endix)  S_av_refresh_elements_range(aTHX_ av, startix, endix)
+STATIC void
+S_av_refresh_elements_range(pTHX_ AV *av, IV startix, IV endix)
+{
+    for(IV ix = startix; ix < endix; ix++) {
+        SV **svp = av_fetch(av, ix, FALSE);
+        SV *newsv = newSVsv_flags(svp ? *svp : &PL_sv_undef,
+                (SV_DO_COW_SVSETSV|SV_NOSTEAL));
+        if(!av_store(av, ix, newsv))
```

Comment:
Is av a TIEARRAY? Is this really going to return false? Why are we continuing execution and not throwing a panic:/heap corruption/OOM? Can it be a TIEARRAY? why not use av_simpleX() family of calls here?

Renew() Newx() handles all OOMs NULL RETVALS for us unless we explicitly temporarily disabled that, which is done approximately in only 1 place inside the entire P5P repo, and ~1-2  places on grep CPAN.